internal/provider/resource_time_sleep_test.go

```diff
@@ -155,20 +159,22 @@ func TestResourceTimeSleepDelete(t *testing.T) {
 }
 
 func TestAccTimeSleep_CreateDuration(t *testing.T) {
-	var time1, time2 string
 	resourceName := "time_sleep.test"
 
+	captureTimeState1 := timetesting.NewExtractState(resourceName, tfjsonpath.New("id"))
+	captureTimeState2 := timetesting.NewExtractState(resourceName, tfjsonpath.New("id"))
```

Comment:
Created https://github.com/hashicorp/terraform-plugin-testing/issues/295

Reply:
I really like the idea of the "multiple value" or "value comparison" checking. We've had a little [discussion](https://github.com/hashicorp/terraform-plugin-testing/pull/276#discussion_r1472542363) around possibly providing helper method(s) that ease extraction of values from state for use in comparing values. Perhaps we could have a 2-phase approach where the convenience function(s) for state value extraction are added into [Adding Built-In State Checks](https://github.com/hashicorp/terraform-plugin-testing/pull/276#top), and subsequently [Consider Multiple Value Checking Interface](https://github.com/hashicorp/terraform-plugin-testing/issues/295#top) for a later PR. This would alleviate the developer burden in terms of having to know about the internals for extracting state values in the first instance. Thoughts?

Reply:
I think it makes sense to tackle this as a follow up to Ben's built-in state check PR, maybe we should consider not deprecating the existing state checks until we have a solid design/answer? We can discuss on https://github.com/hashicorp/terraform-plugin-testing/pull/276#top 👍🏻